app/routes/__auth/login.ts

```diff
@@ -13,23 +13,23 @@ import { generators } from 'openid-client'
 import { getOpenIDClient, oidcStorage, storage } from './auth.server'
 import type { getUser } from './user.server'
 
-function userFromTokenSet(
+export function userFromTokenSet(
   tokenSet: TokenSet
 ): NonNullable<Awaited<ReturnType<typeof getUser>>> {
   const claims = tokenSet.claims()
   const sub = claims.sub
   const email = claims.email as string
-
+  const refreshToken = tokenSet.refresh_token as string
   const idp =
     claims.identities instanceof Array && claims.identities.length > 0
       ? (claims.identities[0].providerName as string)
       : null
-
+  const cognitoUserName = claims['cognito:username'] as string
```

Comment:
Why are we grabbing the username?

Reply:
Ya, this makes more sense when the other function is in mind, but this command needs it:

const command = new AdminAddUserToGroupCommand({
      Username: user.cognitoUserName,
      UserPoolId: process.env.COGNITO_USER_POOL_ID,
      GroupName: 'gcn.nasa.gov/circular-submitter',
    })

Reply:
How is the Cognito username related to the sub?

Reply:
I assume it is tied to the user?  its just how the command expects arguments.

https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminAddUserToGroup.html

Reply:
Right, but how will having that in the session help us? A user won't be adding _themselves_ to a group.

Reply:
Never mind, I understand now.